Info-Orbs/src/widgets/stockScreenWidget.cpp

```diff
@@ -0,0 +1,113 @@
+#include "widgets/stockScreenWidget.h"
+#include <config.h>
+#include <sstream>
+#include <iomanip>
+#include <HTTPClient.h>
+#include <ArduinoJson.h>
+
+StockScreenWidget::StockScreenWidget(ScreenManager &manager) : Widget(manager), currentStockIndex(0), m_stockDelay(6000) {
+}
+
+void StockScreenWidget::setup() {
+    m_stockDelayPrev = 0;
+    stockTickers = {"ABNB", "BRCC", "GOOG", "TSLA", "GME"};
+}
+
+void StockScreenWidget::draw() {
+    // Get initial stock then every m_stockDelay milliseconds
+    if (m_stockDelayPrev == 0 || (millis() - m_stockDelayPrev) >= m_stockDelay) {
+        getStockData();
+        displayStock(currentStockIndex, TFT_WHITE, TFT_BLACK);
+        m_stockDelayPrev = millis();
+        currentStockIndex = (currentStockIndex + 1) % stockTickers.size();
+    }
+}
+
+void StockScreenWidget::update() {
+}
+
+void StockScreenWidget::getStockData() {
+    m_TrackingStock = stockTickers[currentStockIndex];
+    //m_TrackingStock = "ABNB";
+
+    String httpRequestAddress = "https://finnhub.io/api/v1/quote?symbol=" + m_TrackingStock + "&token=c1tjb52ad3ia4h4uee9g";
+
+    HTTPClient http;
+    http.begin(httpRequestAddress);
+    int httpCode = http.GET();
+    
+    if (httpCode > 0) { // Check for the returning code
+        String payload = http.getString();
+        StaticJsonDocument<1024> doc;
```

Comment:
is there an advantage of StaticJsonDocument over simply `JsonDocument doc`?

Reply:
I am not sure about this one. I think it was something I found via Google. I would have to do some research to understand advantages.

Reply:
look at the current WeatherWidget it is using `JsonDocument` and it seems to work. I believe you are statically reserving space where the other is dynamically allocating the space. dynamic is simpler.